### PR TITLE
CT-181-electrum proxy middleware implementation api

### PR DESCRIPTION
**Pull request by Nadin-a**
- added connection param;
- refactored get-merkle route;

### CODE REVIEW

File: src/block.js [hunk 1]
```diff
@@ -14,8 +28,7 @@ router.get('/header', async (req, res) => {
 
 // get current
 router.get('/current', async (req, res) => {
-    const raw = req.query['raw'] === 'true';
```
Comment (Isahara86) [inline src/block.js]: Is this param useless? `const raw = req.query['raw'] === 'true';`

File: src/scripthash.js [hunk 2]
```diff
@@ -1,9 +1,77 @@
...
+    res.json({
+        status: 'success',
+        result: json
+    });
+});
+
+// 'blockchain.scripthash.subscribe'
+router.get('/subscribe', async (req, res) => {
+    const scripthash = req.query['scripthash'];
+    const json = await req.locals.ecl.blockchainScripthash_subscribe(scripthash);
```
Comment (Isahara86) [inline src/scripthash.js]: Is it possible to subscribe with http?

File: src/scripthash.js [hunk 3]
```diff
@@ -1,9 +1,77 @@
...
+    await req.locals.ecl.close();
+
+    res.json({
+        status: 'success',
+        result: json
+    });
+});
+
+// 'blockchain.scripthash.unsubscribe'
+router.get('/unsubscribe', async (req, res) => {
```
Comment (Isahara86) [inline src/scripthash.js]: Is it possible to subscribe with http?

File: src/server.js [hunk 4]
```diff
@@ -65,4 +65,36 @@ router.get('/donation-address', async function (req, res) {
...
+    await req.locals.ecl.close();
+
+    res.json({
+        status: 'success',
+        result: json
+    });
+});
+
+// server.peers.subscribe
+router.get('/peers-subscribe', async function (req, res) {
```
Comment (Isahara86) [inline src/server.js]: Is it possible to subscribe with http?

File: src/block.js (hunk 1 above)
Comment (Nadin-a) [reply src/block.js]: https://electrumx.readthedocs.io/en/latest/protocol-methods.html#blockchain-headers-subscribe

As i see method blockchain.headers.subscribe() doesn't need a parameter

File: src/server.js [hunk 5]
```diff
@@ -65,4 +65,36 @@ router.get('/donation-address', async function (req, res) {
...
+    await req.locals.ecl.close();
+
+    res.json({
+        status: 'success',
+        result: json
+    });
+});
+
+// server.peers.subscribe
+router.get('/get-peers-subscribe', async function (req, res) {
```
Comment (Isahara86) [inline src/server.js]: '/get-peers-subscribe' -  sounds like subscription. How about '/get-peers'

**Pull request merged**